### PR TITLE
More opts to the dispatcher + 1 to the JIT

### PR DESCRIPTION
**Pull request by alyssarosenzweig**

### CODE REVIEW

**Comment by Sonicadvance1** [comment]
Could I appeal for a more descriptive commit message? :)

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/Flags.cpp [hunk 1]
```diff
@@ -202,11 +202,29 @@ OrderedNode *OpDispatchBuilder::GetPackedRFLAG(uint32_t FlagsMask) {
...
+  uint64_t SignBit = (SrcSize * 8) - 1;
+  OrderedNode *Anded = nullptr;
+
+  // OF is set iff the sources have the same sign but the destination sign
+  // differs. If we know a source sign, we can simplify the expression: if
+  // source 2 is known to be positive, we set OF if source 1 is positive and
+  // source 2 is negative. Similarly if source 2 is known negative.
+  uint64_t Const;
+  if (IsValueConstant(WrapNode(Src2), &Const)) {
+    bool Negative = !!(Const & (1ull << SignBit));
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/Flags.cpp:214]: Maybe:

```suggestion
    bool Negative = (Const & (1ull << SignBit)) != 0;
```

which is perhaps a little more straightforward when reading. Just a nit so feel free to ignore

**Comment by alyssarosenzweig** [comment]
> Could I appeal for a more descriptive commit message? :)

Honestly just here for CI, I need to start using Drafts whoops sorry!

**Comment by alyssarosenzweig** [comment]
...and I think that's a wrap for tonight :+1:

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/Flags.cpp [hunk 2]
```diff
@@ -200,13 +200,39 @@ OrderedNode *OpDispatchBuilder::GetPackedRFLAG(uint32_t FlagsMask) {
   return Original;
 }
 
-void OpDispatchBuilder::CalculateOF_Add(uint8_t SrcSize, OrderedNode *Res, OrderedNode *Src1, OrderedNode *Src2) {
+void OpDispatchBuilder::CalculateOF(uint8_t SrcSize, OrderedNode *Res, OrderedNode *Src1, OrderedNode *Src2, bool Sub) {
```
Comment (Sonicadvance1) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/Flags.cpp:203]: The definition in the header here for `bool Sub` is `bool Add`.
Which one is it?

Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/Flags.cpp:203]: Sub, this was a typo. Thanks for spotting that, and fixed!

**Review by Sonicadvance1 — APPROVED** [review]

**Pull request merged**